sdk/src/main/java/com/oursky/authgear/AuthgearCore.kt

```diff
@@ -558,6 +568,208 @@ internal class AuthgearCore(
         )
         val userInfo = oauthRepo.oidcUserInfoRequest(tokenResponse.accessToken)
         saveToken(tokenResponse, SessionStateChangeReason.AUTHENTICATED)
+        disableBiometric()
         return AuthorizeResult(userInfo, uri.getQueryParameter("state"))
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    fun checkBiometricSupported(context: Context, allowed: Int) {
+        requireIsInitialized()
+
+        var allowed = allowed
+        ensureAllowedIsValid(allowed)
+        allowed = convertAllowed(allowed)
+        val result = BiometricManager.from(context).canAuthenticate(allowed)
+        if (result != BiometricManager.BIOMETRIC_SUCCESS) {
+            throw BiometricCanAuthenticateException(result)
+        }
+    }
+
+    fun isBiometricEnabled(): Boolean {
+        requireIsInitialized()
+
+        val kid = this.tokenRepo.getBiometricKeyId(this.name)
+        if (kid == null) {
+            return false
+        }
+        return true
+    }
+
+    fun disableBiometric() {
+        requireIsInitialized()
+
+        val kid = this.tokenRepo.getBiometricKeyId(this.name)
+        if (kid != null) {
+            val alias = "com.authgear.keys.biometric.$kid"
+            removePrivateKey(alias)
+            this.tokenRepo.deleteBiometricKeyId(this.name)
+        }
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    suspend fun enableBiometric(
+        options: BiometricOptions
+    ) {
+        requireIsInitialized()
+
+        val accessToken: String = this.accessToken
+            ?: throw IllegalStateException("enableBiometric required authenticated user.")
```

Comment:
Is there a chance that when user call enableBiometric the access token expired already? It is handled?

Reply:
Not handled. It is actually another problem. Both the iOS SDK and Android has this issue. The handling of refreshing access token is not automatic. That's why I proposed to improve the SDK before soft-launch.